src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker_Generators.cs

```diff
@@ -281,6 +283,7 @@ private partial class CompilationTracker : ICompilationTracker
             }
 
             using var generatedDocumentsBuilder = TemporaryArray<SourceGeneratedDocumentState>.Empty;
+            var generationDateTime = DateTime.Now;
```

Comment:
💡 Comment here to make sure it's not accidentally moved inside the `foreach` loop later. We want all files from a generation pass to have the same stamp, even though they were not all produced at exactly the same time.

Reply:
sure :)